### ISSUE TITLE
Android support broken

### ISSUE THREAD

**Issue by codyhazelwood** [oblador/react-native-keychain#21]
Something odd must have happened during the commit squashing / merging process, because Android support is not working for the v0.3.0 release.  iOS support works fine.

If I revert back to my branch git://github.com/codyhazelwood/react-native-keychain.git#8a2e5ab12aed2cfc5f4a23ffeec2dad56b017a48, everything works as expected.

**Comment by oblador** [comment]
@vonovak: care to look at this?

**Comment by codyhazelwood** [comment]
The code that causes the issue:
```javascript
const Config = {portalUrl: 'https://www.myurl.com/'};
...
    componentDidMount() {
        ....
        Keychain.getInternetCredentials(Config.portalUrl)
            .then(credentials => {
                let {username: email, password} = credentials;
                this.setState({email, password});
            });
    }

    _signIn() {
        Auth.signIn(this.state.email, this.state.password)
            .then(user => {
                if (user) {
                    ....
                    // Save login info
                    Keychain.setInternetCredentials(
                        Config.portalUrl,
                        this.state.email,
                        this.state.password
                    ).catch(err => console.error(err));
                }
            })
    }
```

Not sure that these logs are too helpful, but here they are.

Logs in v0.3.0 for `getInternetCredentials`:
```
05-31 18:37:26.469 23797 24840 D KeychainModule : will attempt to decrypt for https://www.myurl.com/AQL2cswUfqA18qpsjApUJdf5tlJnM/B0eEWQ2XirtXDSlLI+SrdHKH3mYlk4idiAelmwWgM=
05-31 18:37:26.469 23797 24840 D KeychainModule : :AQJd2LLAUciJghFPh53t6f41Kdk7xt2J0tqc5ZSnB41QsEVLYqw=
05-31 18:37:26.512 23797 24839 I ReactNativeJS: 'getInternetCredentials - likely not a critical error', 'convertError@http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:112986:17\nhttp://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:112883:20\n__invokeCallback@http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:6760:15\nhttp://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:6643:24\nguard@http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:6588:3\ninvokeCallbackAndReturnFlushedQueue@http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:6642:6\ninvokeCallbackAndReturnFlushedQueue@[native code]\nglobal code'
05-31 18:37:26.513 23797 24839 E ReactNativeJS: [Error: Error]
05-31 18:37:26.520 23797 24840 E unknown:React: Error, stack:
05-31 18:37:26.520 23797 24840 E unknown:React: convertError@112986:17
05-31 18:37:26.520 23797 24840 E unknown:React: <unknown>@112883:20
05-31 18:37:26.520 23797 24840 E unknown:React: __invokeCallback@6760:15
05-31 18:37:26.520 23797 24840 E unknown:React: <unknown>@6643:24
05-31 18:37:26.520 23797 24840 E unknown:React: guard@6588:3
05-31 18:37:26.520 23797 24840 E unknown:React: invokeCallbackAndReturnFlushedQueue@6642:6
```

Logs in my branch for `getInternetCredentials`:
```
05-31 19:00:00.737 31125 31180 D KeychainModule : will attempt to decrypt for https://www.myurl.com/AQJ4cojg/0e7HOtsRCuvSELsHD2dCNBE/Bu5U9WBuFN/Aj1L74bS/A9b5tUTuX6kFEuLDIQ=
05-31 19:00:00.737 31125 31180 D KeychainModule : :AQJznomEHe5kav/afbJUxc5u+suv47bzc/3gjwyn8zl3XM5w6sI=
```

**Comment by vonovak** [comment]
bummer @codyhazelwood this happens when no credentails are saved or are they there and it fails when decrypting?

**Comment by vonovak** [comment]
@codyhazelwood do you have the libconceal libraries in lib directory?

**Comment by vonovak** [comment]
@codyhazelwood did you test after changing
```
  if (Platform.OS === 'android') {
      return err;
  }
```
 to
```
  if (Platform.OS === 'android') {
      return new Error(err);
  }
```
? It works for me when I revert the change, i.e. the first snippet works.

@oblador https://github.com/oblador/react-native-keychain/blob/master/index.js#L44 any reason why convertError is called twice?

**Comment by oblador** [comment]
@vonovak: no that looks like a bug to me.

**Comment by oblador** [comment]
This module is pretty ripe for a refactor tbh :-) Would like to use native promise support instead for example which would make the codebase cleaner, do automatic error conversion and even provide a native stack trace. Will break some backwards compatibility though, but likely not an issue.

**Comment by vonovak** [comment]
@codyhazelwood try the index.js file from my android branch.. it works for me.

**Comment by codyhazelwood** [comment]
Sorry, I just noticed when I was testing that I had locked my package.json to the commit before, without the error handling change.  @vonovak your latest index.js works fine.

**Comment by vonovak** [comment]
PR #22

**Comment by oblador** [comment]
Released as 0.3.1
